### PR TITLE
Marshal SizeOf Refactoring

### LINKED ISSUE #5280: 'Marshal.SizeOf(Type)' is obsolete errors in MG source.
Full error detail
Warning	CS0618	'Marshal.SizeOf(Type)' is obsolete: 'SizeOf(Type) may be unavailable in future releases. Instead, use SizeOf<T>(). For more info, go to http://go.microsoft.com/fwlink/?LinkID=296515'	MonoGame.Framework.WindowsUniversal	\MonoGame.Framework\Graphics\TextureCube.DirectX.cs	105	

If no one else is looking at this, I'll do a quick PR as the fix is quite easy.  Will just need to validate if the fix is applicable to all platforms.

### PR DESCRIPTION
**Pull request by DDReaper**
Split the ReflectionHelpers out in to a partial class implementation
Update based on #5316
Fixes #5280

### CODE REVIEW

**Comment by DDReaper** [comment]
Over to you @tomspilman to check with Console builds

**Comment by tomspilman** [comment]
Yeah this will work... it allows me to pick the right file on XB1/Vita/etc based on what is available there.

Thanks @DDReaper !

File: MonoGame.Framework/Utilities/ReflectionHelpers.Default.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        /// <summary>
+        /// Generics handler for Marshal.SizeOf
+        /// </summary>
+        internal static class SizeOf<T>
+        {
+            static int _sizeOf;
+
+            static SizeOf()
+            {
+                _sizeOf = Marshal.SizeOf<T>();
```
Comment (tomspilman) [inline MonoGame.Framework/Utilities/ReflectionHelpers.Default.cs]: You know I was thinking that `Marshal.SizeOf<T>()` would internally do the same optimization we're doing here.  But actually it doesn't:

https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Runtime/InteropServices/Marshal.cs#L256

I don't see why they even made this change really... seems pointless.

Comment (DDReaper) [inline MonoGame.Framework/Utilities/ReflectionHelpers.Default.cs]: I did wonder why they were changing the API as it doesn't seem to offer any improvement.
However the old method is now marked as deprecated.

**Pull request merged**